examples/platforms/nrf528xx/src/radio.c

```diff
@@ -971,3 +1076,30 @@ uint32_t nrf_802154_random_get(void)
 {
     return otRandomNonCryptoGetUint32();
 }
+
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+void otPlatRadioSetMacKey(otInstance *    aInstance,
+                          uint8_t         aKeyIdMode,
+                          uint8_t         aKeyId,
+                          const otMacKey *aPrevKey,
+                          const otMacKey *aCurrKey,
+                          const otMacKey *aNextKey)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    OT_UNUSED_VARIABLE(aKeyIdMode);
+
+    assert(aPrevKey != NULL && aCurrKey != NULL && aNextKey != NULL);
```

Comment:
Do we ensure that Radio is in the Sleep state while we change the keys? Otherwise, we may experience some race conditions (e.g. when encrypting the ACK). The alternative would be to put this code into the critical section.

Reply:
Yes, setting MAC key will be a problem if ACK is generated in interrupt. I would prefer putting it into critical sections rather than deferring the setting until radio is in sleep state.

So this means I have to expose `nrf_802154_critical_section_enter` and `nrf_802154_critical_section_exit` to radio.c